src/charm.py

```diff
@@ -366,9 +365,7 @@ def _on_collect_status(self, event: CollectStatusEvent):  # noqa: C901
             return
         role_id, secret_id = self._get_approle_auth_secret()
         if not role_id or not secret_id:
-            event.add_status(
-                BlockedStatus("Please authorize charm (see `authorize-charm` action)")
-            )
+            event.add_status(BlockedStatus("Please grant charm a token secret"))
```

Comment:
I would keep the `Please authorize charm` message.
Maybe changing the `see authorize-charm action` to `use juju user secrets`

Reply:
changed the whole thing. please look again.